pages/ownerInformationPage.ts

```diff
@@ -84,26 +84,36 @@ export class OwnerInformationPage extends HelperBase{
 
     }
 
-    async validateOwnerInformation(firstName: string, lastName: string, address: string, city: string, telephone: string){       
+    async validateOwnerInformation(fullName: string, address: string, city: string, telephone: string){       
         const ownerTable = this.page.getByRole('table').filter({hasText: 'Address'})
 
-        await expect(ownerTable.getByRole('row', {name: 'Name'})).toContainText(`${firstName} ${lastName}`)
+        await expect(ownerTable.getByRole('row', {name: 'Name'})).toContainText(fullName)
         await expect(ownerTable.getByRole('row', {name: 'Address'})).toContainText(address)
         await expect(ownerTable.getByRole('row', {name: 'City'})).toContainText(city)
         await expect(ownerTable.getByRole('row', {name: 'Telephone'})).toContainText(telephone)
         
     }
 
-    async validatePetNamesArePresentFromListOfPetNames(petNames: string[]){
-        const allPetInformationTables = await this.page.locator('table app-pet-list').all()
+    async validatePetNamesArePresentFromListOfPets(pets: {}[]){
+        const petsAndVistsTables = await this.page.locator('table app-pet-list').all()
 
+        // Extracting all pet names from API Mock into a list
+        const petNames: string[] = []
+        for(let pet of pets){
+            petNames.push(pet.name)
+        }
```

Comment:
One more refactoring and simplification you can do.
Instead of processing the pets object and extracting the list of names from each pet using the loop, you can use a single line JS expression using a `map` and get this list from the object and pass this object as the argument. It will be a cleaner approach, because more understandable, what kind of data the method expect as an argument.

So in the test you can replace this: `const pet = owner?.pets.find(pet => pet.id === 1)`
With this: `const petNames = owner.pets.map(pet => pet.name);`

And just pass the petNames array as an argument to the method.